inc/napa/async-sender.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <mutex>
+#include <queue>
+#include <atomic>
+#include <uv.h>
+
+namespace napa{
+
+class AsyncTaskSender {
+public:
+    typedef std::function<void(void)> AsyncTask;
+
+    static AsyncTaskSender* Create(uv_loop_t* loop) {
+        return new AsyncTaskSender(loop);
+    }
+
+    void stop() {
+        needStop_ = true;
+        uv_async_send(&handle_);
+    }
+
+    void post(AsyncTask task) {
+        if (!needStop_) {
+            std::lock_guard<std::mutex> lock(mutex_);
+            queue_.emplace(std::move(task));
+            uv_async_send(&handle_);
+        }
```

Comment:
Is it better to add an assert when a stopped AsyncTaskSender try to post a task?